api/tests/unit/features/workflows/core/test_unit_workflows_models.py

```
@@ -1283,3 +1257,42 @@ def test_change_request_commit__v1_segment_override_draft__inherits_mv_hashing_s
+
+    # When / Then
+    with pytest.raises(
+        ValueError, match="System segments cannot be overwritten via change request drafts."
+    ):
+        change_request.commit(admin_user)
+
+    # Assert that the transaction rolled back successfully
+    feature_state.refresh_from_db()
+    change_request.refresh_from_db()
+    assert feature_state.version == initial_version
+    assert change_request.committed_at is None
```

Ruff (0.16.1)

[warning] No newline at end of file

Add trailing newline

(W292)